### ISSUE TITLE
Boarding is unreliable

### ISSUE THREAD

**Issue by kilobyte** [endless-sky/endless-sky#42]
In 0.7.10, pressing the 'b' key once was enough.  In 0.7.11, though, there is only a small chance it will actually board, most of the time it comes to the target then abandons its efforts.  Only after repeatedly trying (easiest by holding down 'b') I can get to board a ship.

**Comment by endless-sky** [comment]
Boarding is aborted if your ship takes damage during the boarding process. Were you being fired on when this happened? I've been doing some play testing today and didn't encounter any issues with boarding.

**Comment by kilobyte** [comment]
I tested it a bunch more times (hard to disable things among the Hai...), and the trouble is reproducible every time.  Perhaps it might be something related to aftereffects of being hit by Hai weaponry staying way longer than the visual effect?  I can try on ordinary pirates if you wish.

**Comment by endless-sky** [comment]
The latest change I've checked in (91171c4) makes it so that taking damage no longer causes your ship to abort the boarding process. Could you test that out and see if this bug goes away? (I've been unable to reproduce the bug, including against Hai ships, so I can't test it myself. Is it possible that they were overheated rather than truly disabled, and they were "coming back to life" while you were trying to board them?) If that change doesn't fix it, could you attach your pilot file so I can see if I can reproduce the bug on my system with your pilot? Thanks!

**Comment by kilobyte** [comment]
Doesn't seem to help.  It's random, sometimes boarding succeeds on the first try, some times you need to hold 'b' for quite a while while the ship wiggles around.

My save file: http://angband.pl/tmp/Meow%20Mix.txt

**Comment by endless-sky** [comment]
Thanks for the save file! I gave it a try and found that if "automatic aiming" and "automatic firing" are both turned on, the boarding is not working right. If I turn "automatic firing" off (leaving "automatic aiming" on), boarding works fine. Does that match what you're seeing?

**Comment by endless-sky** [comment]
If the automatic firing was indeed what was causing is your issue, see if commit ac1c3f8 fixes it.

**Comment by kilobyte** [comment]
Yes, this fixes it.

I guess 91171c4 is no longer needed then.
